### ISSUE TITLE
Entity.BuildInput is not called on pawns controlled by a bot

### ISSUE THREAD

**Issue by chrisspieler** [sboxgame/issues#2879]
### Describe the bug

tl;dr: The XML docs of Entity.BuildInput erroneously claim that BuildInput is called serverside for pawns controlled by bots, when actually it seems to never be called at all for bots. Are the docs wrong, or is this a bug?

I am currently working on developing bots that control pawns and move around. According to example shown on the [wiki](https://wiki.facepunch.com/sbox/Bots), bot inputs are simulated by setting stuff in the Input class. The relevant portion of the example is:

```csharp
public override void BuildInput()
{
    // Here we can choose / modify the bot's input each tick.
    // We'll make them constantly attack by holding down the PrimaryAttack button.
    Input.SetButton( InputButton.PrimaryAttack, true );
}
```
There's nothing on the wiki page that says how bot movement and look inputs are set, so I looked at other members of Input to see what I could find. I see that Input.AnalogLook and Input.AnalogMove both have public setters. So, I set these inputs in Bot.BuildInput with the expectation that, because my pawn uses these values in Entity.BuildInput to move and look, this way of doing things will automatically work with my pawn.

Now, Bot.BuildInput is called serverside, as stated by its XML comments. But this isn't an issue in my case, because according to the XML comments for Entity.BuildInput:

"Pawns get a chance to mess with the input. This is usually called on the client, but for bots it will be called on the server."

Anyway, I run my game expecting for my bots to move, but they don't. I eventually add a line in my pawn's BuildInput method to log to the console and find that apparently Entity.BuildInput is not being called at all for pawns that are controlled by my bot.

As an aside, I think it would be helpful to confirm what the actual intended means of simulating bot movement and look inputs is (i.e., should I really be setting AnalogLook and AnalogMove, or is there something else I should be doing?)

### To Reproduce

1. Instantiate a Bot
2. Ensure that the client of the bot is assigned a pawn with code in its BuildInput method that logs to the console
3. Check whether the bot's pawn has actually logged to console

### Expected behavior

Because BuildInput is supposed to be invoked on the bot's pawn, you'll see a message logged in console for this bot.

**Comment by kurozael** [comment]
This actually is a documentation error. A pawn's BuildInput shouldn't be called at all for a bot, you use Bot.BuildInput for this purpose.

You shouldn't need to be setting Input.AnalogMove or AnalogLook in this case because from Bot.BuildInput you can directly set the input direction of the pawn.

Why does the Bot class have its own build input? That's because unlike with players bots do not actually have inputs, you simply decide how they move or behave. This doesn't mean you can't share a pawn entity between a bot and a real player, but it does mean they can't share the same BuildInput logic.

**Comment by chrisspieler** [comment]
Thanks for the response, @kurozael ! 

This clears things up for me. I like the idea of having a generic bot that does not necessarily have knowledge of specific pawn implementations, so having the BuildInput of Bot set properties of Input and then call the BuildInput of the pawn's Entity suits me just fine. I just updated my code and tested that it works fine for me. I'll go ahead and update the wiki with this new information. 

 The only remaining issue is inaccurate XML docs for Entity.BuildInput. I don't know whether that means I should close this issue; I'll defer to y'all.
